CIP-????/README.md

```diff
@@ -0,0 +1,250 @@
+
+- The question and answers
+- The index of the chosen answer from the available choices
+- A signature from the SPO's current VRF credentials or Ed25519 declared cold credentials.
+
+> **Note**
+> In this document, every time we refer to a _serialized object_, we refer to its **canonical** CBOR representation. In particular, keys in a map are always ordered alphabetically.
+
+### Question structure
+
+A question is posted in a transaction's metadata using the metadata label `94` and the following metadata structure:
+
+```cbor
+question =
+  { 0: prompt
+  , 1: [ * choice ]
+  , ? "_": nonce
+  }
+
+prompt =
+  text .size (0..64) / [ * text .size (0..64) ]
+
+choice =
+  text .size (0..64) / [ * text .size (0..64) ]
+
+nonce =
+  uint
+```
+
+A nonce is optionally included to provide non-replayability should the same question and answers be asked multiple times over different periods. The transaction carrying a question **must** be signed by one of the genesis delegate keys to be considered valid. This genesis key signature isn't captured in the metadata but in the transaction itself as an extra signatory.
+
+For example, in [CBOR-diagnostic](https://www.rfc-editor.org/rfc/rfc8949#name-diagnostic-notation) notation:
+
+```cbor
+{ 94:
+  { 0: "Pineapples on pizza?"
+  , 1:
+    [ "yes"
+    , "no"
+    ]
+  }
+}
+```
+
+Which serialises down to:
+
+```
+A1185EA2007450696E656170706C6573206F6E2070697A7A613F018263796573
+626E6F
+```
+
+### Answer structure
+
+Similarly, an answer to a question is posted as transaction's metadata using the label `94` and the following metadata structure:
+
+```cbor
+answer =
+  { 2: question_hash
+  , 3: choice
+  , ? 4: vrf_witness
+  , ? 5: ed25519_witness
+  }
+
+question_hash =
+  bytes .size 32
+
+vrf_witness =
+  [ $vrf_vkey, vrf_proof ]
+
+vrf_proof =
+  [ bytes .size 64, bytes .size 16 ]
+
+ed25519_witness =
+  [ $vkey, $signature ]
+```
+
+Some remarks:
+
+1. The field `2` (`question_hash`) is a blake2b-256 hash digest, whose preimage is the entire serialised question metadata payload (with the `94` top-level label).
+1. The fields `4` and `5` are not mutually optional; said differently, at least one of the two must be provided.
+1. The `$vrf_vkey`, `$vkey` and `$signature` are as defined in the original [Cardano ledger specifications](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/real/crypto.cddl) with the usual rules applying to transaction metadata (i.e. byte string longer than 64 bytes are encoded as lists)
+1. In either case (VRF or Ed25519), the witness' signature is calculated from the blake2b-256 hash digest of the serialised `answer` restricted to fields `2` and `3`.
+
+For example, in [CBOR-diagnostic](https://www.rfc-editor.org/rfc/rfc8949#name-diagnostic-notation) notation:
+
+```cbor
+{
+  94: {
+    2: h'3d4854aed3e31d0f5b6637fcd73e8713d9eee86a81d20598f6cb88d9b2bc0a39',
+    3: 2,
```

Comment:
Is this example based on the `question` example above where two string values are provided as the `choices`? If so, should the entries in `choices` be serialized based on a 1-index of the choices? As defined a "choice" should be a string or an array of strings but here is presented as a uInt value?

Reply:
It's a the index (0-based) of the selected choice from the corresponding question indeed. There's no need to replicate the full answer as this takes a needless amount of space (times 3000 for each pool).

Reply:
I'll clarify that bit in the proposal.